server/index.js

```diff
@@ -67,6 +67,111 @@ app.get('/api/products/:productId', (req, res, next) => {
     });
 });
 
+app.get('/api/cart/', (req, res, next) => {
+  if (typeof req.session.cartId === 'undefined') {
+    res.json([]);
+  } else {
+    const sql = `select "c"."cartItemId",
+      "c"."price",
+      "p"."productId",
+      "p"."image",
+      "p"."name",
+      "p"."shortDescription"
+      from "cartItems" as "c"
+      join "products" as "p" using ("productId")
+      where "c"."cartId" = $1`;
+    const values = [req.session.cartId];
+    return db.query(sql, values)
+      .then(result => {
+        const cartItem = result.rows[0];
+        const newArr = [];
+        newArr.push(cartItem);
+        return res.json(newArr);
+      });
+  }
+});
+
+app.post('/api/cart/', (req, res, next) => {
+  const productId = req.body.productId;
+  if (productId < 0) {
+    res.status(400).json({
+      error: 'Invalid productId'
+    });
+    return;
+  }
+  const text = `
+    select "price"
+    from "products"
+    where "productId" = $1
+    `;
+  const values = [productId];
+  db.query(text, values)
+    .then(result => {
+      const productRows = result.rows;
+      const price = result.rows[0].price;
+      if (!productRows) {
+        res.status(400).json({
+          error: 'No data to return'
+        });
+      }
+      if (typeof req.session.cartId === 'undefined') {
+        const sql = `insert into "carts" ("cartId", "createdAt")
+        values (default, default)
+        returning "cartId"`;
+        return db.query(sql)
+          .then(result => {
+            const cartId = result.rows[0].cartId;
+            return { cartId, price };
+          });
+      } else {
+        const sql = `select "cartId"
+         from "carts"
+         where "cartId" = $1`;
+        const values = [req.session.cartId];
+        return db.query(sql, values)
+          .then(result => {
+            const cartId = result.rows[0].cartId;
+            return { cartId, price };
+          });
```

Comment:
In this `else`, you already know the `cartId` is in `req.session`, so you don't have to query the database for it.